### PR TITLE
feat: Add CanSpawnMobile overload with props Z-range support.

### PR DESCRIPTION
**Pull request by kamronbatman**
### Summary

- Adds CanSpawnMobile(x, y, minZ, maxZ, canSwim, cantWalk, out spawnZ) overload for finding spawn surfaces within a Z range
- Adds CanSpawnItem(x, y, minZ, maxZ, out spawnZ) for item spawning with Surface+Impassable support (tables, furniture)
- Uses bitmask optimization inspired by Item.DropToWorld's m_OpenSlots pattern for O(1) surface/blocker checks
- HomeRange spawners now use surface detection to set proper Z bounds

### Key Changes

Map.cs:
- CanSpawnMobile with Z-range finds lowest valid surface for mobiles
- CanSpawnItem with Z-range finds lowest valid surface for items (including tables)
- CanFitItem for point-check item placement on Surface+Impassable tiles
- Bitmask approach eliminates nested loops and stackalloc arrays

Spawners:
- Simplified GetSpawnPosition using new Z-range methods
- HomeRange setter detects surface below spawner for proper Z bounds
- Consistent handling for mobiles and items

### Bug Fixes

- Water tiles (Impassable | Wet) no longer block swimming mobs
- Items can now spawn on tables/furniture (Surface+Impassable)

### Test Plan

- Run dotnet test - 631 tests pass
- Manual testing: multi-story spawning, water mobs, item spawning on tables
- Verify HomeRange spawner movement shifts bounds correctly

### CODE REVIEW

File: Projects/Server/Maps/Map.cs [hunk 1]
```diff
@@ -922,6 +922,114 @@ public bool CanFit(
...
+            }
+        }
+
+        if (found)
+        {
+            spawnZ = bestZ;
+        }
+
+        return found;
+    }
```
[the remark this answers is not in the document]
Comment (kamronbatman) [reply Projects/Server/Maps/Map.cs:1363]: Here's an analysis of our test coverage for those scenarios:

  | Scenario                                                     | Coverage   | Test(s)                                                                                                          |
  |--------------------------------------------------------------|------------|------------------------------------------------------------------------------------------------------------------|
  | Correct Z selection for land tiles within range              | ✅ Covered | EmptyLocationWithLandReturnsTrue, ZRangeExcludingLandFails                                                       |
  | Proper handling of wet vs non-wet surfaces for swimming mobs | ✅ Covered | WaterSurfaceForSwimmingMob (canSwim=true finds water)                                                            |
  | Water-only mob validation (cantWalk=true)                    | ✅ Covered | CantWalkSkipsLandSurface (cantWalk=true rejects land)                                                            |
  | Surface calculation for static tiles and world items         | ✅ Covered | FindsSurfaceOnMulti, WorldItemSurfaceWorks                                                                       |
  | Lowest surface preference when multiple valid surfaces exist | ✅ Covered | FindsLowestValidSurface, MalasBuilding_LowestFloorPreferred                                                      |
  | Boundary cases (minZ, maxZ, out of range)                    | ✅ Covered | ZRangeExcludingLandFails, ZRangeSelectsCorrectFloor, BlockerOutsideZRangeIgnored, InvalidCoordinatesReturnsFalse |

  Note on "Highest surface preference": Copilot may be confused here. Our implementation correctly prefers the lowest valid surface (ground floor over upper floors), which is the expected behavior for spawning - you want mobs at ground level, not floating on upper floors. This is tested by FindsLowestValidSurface and MalasBuilding_LowestFloorPreferred.

  Verdict: The claim appears to be incorrect - we have coverage for all these scenarios. The tests cover:
  - 19 total CanSpawnMobile tests (8 basic + 11 TileData-dependent)
  - Land, multi, and world item surfaces
  - Swimming/walking mob variations
  - Z-range boundary conditions
  - Multi-floor buildings with lowest-floor preference
